samples/javascript_nodejs/56.teams-file-upload/services/fileService.js

```diff
@@ -0,0 +1,43 @@
+const fs = require('fs');
+const util = require('util');
+const axios = require('axios');
+const readdir = util.promisify(fs.readdir);
+
+// Generates File Name with Sequence
+const geneFileName = async (fileDir) => {
+    const filenameConst = 'UserAttachment';
+    const files = await readdir(fileDir);
+    const filteredFiles = files.filter(f => f.includes(filenameConst)).map(f => parseInt(f.split(filenameConst)[1].split('.')[0]));
+    let maxSeq = 0;
+    if (filteredFiles.length > 0) {
+        maxSeq = Math.max.apply(Math, filteredFiles);
+    } else {
+        maxSeq = 0;
+    }
+    const filename = `${ filenameConst }${ maxSeq + 1 }.png`;
+    return filename;
+};
+
+// Download and Save Streams into File
+const writeFile = async (contentUrl, config, filePath) => {
```

Comment:
I think (depending on axios version) this should work:

```javascript
const response = await axios({ method: 'GET', url: contentUrl, responseType: 'stream' });
await new Promise((resolve, reject) => response.pipe(fs.createWriteStream(filePath)).once('finish', resolve).once('error', reject));
```

It's a little nicer as there is less nesting and also ensures that errors are propagated.